src/useMediaQuery.ts

```diff
@@ -20,7 +20,7 @@ const useMediaQuery = (mediaQuery: string) => {
     return false
   }
 
-  const [isVerified, setIsVerified] = useState(!!window.matchMedia(mediaQuery).matches)
+  const [isVerified, setIsVerified] = useState(false)
```

Comment:
Mh, I'm not 100% sure this is the correct way to fix this issue, cause this will return a false negative of the first render. 
I think the correct way to fix this should be something like:

```ts
const [isVerifed, setIsVerified] = useState(isClient() ? !!window.matchMedia(mediaQuery).matches : false)
```

On top of that the `mediaQuery ` dependency of the `useEffect` is actually correct, we should not remove it... 

probably the ternary operator for the default value of `isVerified` fixes it. 

Keep me posted

Reply:
I will get more research because we will back then to the previous code because the top of the file we are checking if it is a client so this condition does not make sense

Reply:
I think we can close this PR since this warning is triggered in the old next.js and the problem was with checking the window object. But I run this hook in my new next.js app and I can't even reproduce this